src/GraphWebhooks/Graph/ChangeNotificationEncryptedContentExtensions.cs

```diff
@@ -0,0 +1,91 @@
+        var rootNode = parseNodeFactory.GetRootParseNode("application/json", contentStream);
+        return rootNode.GetObjectValue<T>((parsable) => new T());
+    }
+
+    private static async Task<string> DecryptAsync(
+        this ChangeNotificationEncryptedContent encryptedContent,
+        Func<string, string, Task<X509Certificate2>> certificateProvider)
+    {
+        if (certificateProvider == null)
+            throw new ArgumentNullException(nameof(certificateProvider));
+
+        _ = encryptedContent.EncryptionCertificateId ??
+            throw new Exception("Certificate ID missing in encrypted content");
+        _ = encryptedContent.EncryptionCertificateThumbprint ??
+            throw new Exception("Certificate thumbprint missing in encrypted content");
+        _ = encryptedContent.DataKey ??
+            throw new Exception("Data key missing in encrypted content");
+        _ = encryptedContent.Data ??
+            throw new Exception("Data missing in encrypted content");
+
+        using var certificate = await certificateProvider(encryptedContent.EncryptionCertificateId, encryptedContent.EncryptionCertificateThumbprint).ConfigureAwait(false);
+        using var rsaPrivateKey = certificate.GetRSAPrivateKey() ??
+            throw new Exception("Could not get RSA private key from certificate");
+        var decryptedSymmetricKey = rsaPrivateKey.Decrypt(Convert.FromBase64String(encryptedContent.DataKey), RSAEncryptionPadding.OaepSHA1);
+        using var hashAlg = new HMACSHA256(decryptedSymmetricKey);
+        var expectedSignatureValue = Convert.ToBase64String(hashAlg.ComputeHash(Convert.FromBase64String(encryptedContent.Data)));
+        if (!string.Equals(encryptedContent.DataSignature, expectedSignatureValue))
+        {
+            throw new InvalidDataException("Signature does not match");
+        }
+        else
+        {
+            return Encoding.UTF8.GetString(AesDecrypt(Convert.FromBase64String(encryptedContent.Data), decryptedSymmetricKey));
+        }
+    }
+
+    private static byte[] AesDecrypt(byte[] dataToDecrypt, byte[] key)
+    {
+        try
+        {
+            #pragma warning disable SYSLIB0021
+            using var cryptoServiceProvider = new AesCryptoServiceProvider
+            {
+                Mode = CipherMode.CBC,
+                Padding = PaddingMode.PKCS7,
+                Key = key
+            };
+            #pragma warning restore SYSLIB0021
+            var numArray = new byte[16]; //16 is the IV size for the decryption provider required by specification
+            Array.Copy(key, numArray, numArray.Length);
+            cryptoServiceProvider.IV = numArray;
+            using var memoryStream = new MemoryStream();
+            using var cryptoStream = new CryptoStream(memoryStream, cryptoServiceProvider.CreateDecryptor(), CryptoStreamMode.Write);
+            cryptoStream.Write(dataToDecrypt, 0, dataToDecrypt.Length);
```

Comment:
I think we need to use the writeAsync method instead otherwise the runtime is going to yell at people by default, also we should pass the cancellation token

Reply:
Core SDK should get fixed too